### PR TITLE
Merge LCS into Master

### PR DESCRIPTION
**Pull request by tomlazar**

### CODE REVIEW

File: Algrothms/String/LongestCommonSubstring/LongestCommonSubstring.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+    // Fields
+    final static boolean DEBUG = true;
+
+    // LCS Parameters
+    private int MatchBonus;
+    private int MisMatchPenalty;
+    private int InsertionPenalty;
+    private int DeletionPenalty;
+    private boolean GlobalAllignment;
```
Comment (Minnow42) [inline Algrothms/String/LongestCommonSubstring/LongestCommonSubstring.java]: Misspelled alignment

File: Algrothms/String/LongestCommonSubstring/LongestCommonSubstring.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+    //Constructors
+
+    /**
+     * Creates a LCS object with default score values, this can be used
```
Comment (Minnow42) [inline Algrothms/String/LongestCommonSubstring/LongestCommonSubstring.java]: Run-on sentence

File: Algrothms/String/LongestCommonSubstring/LongestCommonSubstring.java [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+                                  boolean globalAllignment) {
+        MatchBonus = matchBonus;
+        MisMatchPenalty = misMatchPenalty;
+        InsertionPenalty = insertionPenalty;
+        DeletionPenalty = deletionPenalty;
+        GlobalAllignment = globalAllignment;
+    }
+
+    /**
+     * Runs the program all interactive like.
```
Comment (Minnow42) [inline Algrothms/String/LongestCommonSubstring/LongestCommonSubstring.java]: Clarify this please

File: Algrothms/String/LongestCommonSubstring/LongestCommonSubstring.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+                System.out.println("FIRST SEQUENCE: " + S);
+                System.out.println("SECOND SEQUENCE: " + T);
+            }
+
+            // ** PROCESS TABLE **
+            String res = lcs.lcs(S, T);
+            System.out.println(res);
+
+            // Decide next Program
+            System.out.print("Do you want to try again (y/n) ");
```
Comment (Minnow42) [inline Algrothms/String/LongestCommonSubstring/LongestCommonSubstring.java]: Missing a question mark

File: Algrothms/String/LongestCommonSubstring/README.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+**Language:** Java   
+**Author:** Tom L.    
+**Date:** 9-29-2016   
+**Version:** v0.1.0   
+
+## Background
+https://en.wikipedia.org/wiki/Longest_common_subsequence_problem
+
+Used commonly in bioinformatics to find common patterns in many sequences of DNA, there are several different variations on this but they can all be built using the same basis.
```
Comment (Minnow42) [inline Algrothms/String/LongestCommonSubstring/README.md]: Another run-on

File: Algrothms/String/LongestCommonSubstring/README.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+## Complexity
+
+This is a Dynamic Programming implementation of the Longest Common subsequence Problem.
+
+It has a time Complexity of O(nm), and a space complexity of O(nm), further optimizations could be used to improve the space efficiency to O(n + m)
```
Comment (Minnow42) [inline Algrothms/String/LongestCommonSubstring/README.md]: Run-on

**Review by Minnow42 — CHANGES_REQUESTED** [review]
See comments

File: Algrothms/String/LongestCommonSubstring/LongestCommonSubstring.java (hunk 2 above)
Comment (tomlazar) [reply Algrothms/String/LongestCommonSubstring/LongestCommonSubstring.java]: WHO CARES

**Review by Minnow42 — APPROVED** [review]
ok sure tommy

**Pull request merged**